homeassistant/components/freedompro/lock.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Freedompro lock."""
+import json
+
+from pyfreedompro import put_state
+
+from homeassistant.components.lock import LockEntity
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro lock."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "lock"
+    )
+
+
+class Device(CoordinatorEntity, LockEntity):
+    """Representation of an Freedompro lock."""
+
+    def __init__(self, hass, api_key, device, coordinator):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Pass in the aiohttp client session instead.